controllers/nodehealthcheck_controller.go

```diff
@@ -459,7 +469,20 @@ func (r *NodeHealthCheckReconciler) remediate(node *v1.Node, nhc *remediationv1a
 	}
 
 	// create remediation CR
-	created, err := rm.CreateRemediationCR(remediationCR, nhc)
+	created, leaseRequeueIn, err := rm.CreateRemediationCR(remediationCR, nhc)
+
+	//An unhealthy node exist but remediation couldn't be created because lease wasn't obtained - update unhealthy nodes and requeue
+	if _, isLeaseAlreadyTaken := err.(lease.AlreadyHeldError); isLeaseAlreadyTaken {
+		resources.UpdateStatusNodeUnhealthy(node, nhc)
+		return leaseRequeueIn, nil
+	}
+
+	//Lease is overdue
+	if _, isLeaseOverDue := err.(resources.LeaseOverDueError); isLeaseOverDue {
+		resources.UpdateStatusNodeUnhealthy(node, nhc)
+		return nil, r.addTimeOutAnnotation(rm, remediationCR, metav1.Time{Time: currentTime()})
```

Comment:
I think we should update the NHC CR status here as well, similar to the `startedRemediation.TimedOut = &now` at the end of this method. Copying the `startedRemediation := ...` call to here should be fine, moving it above this block won't work.